coast/gridded.py

```diff
@@ -0,0 +1,865 @@
+from .COAsT import COAsT
+from .config import ConfigParser
+from .logging_util import get_slug, debug, info, warn, error
+
+
+class Gridded(COAsT):  # TODO Complete this docstring
+    """
+    Words to describe the NEMO class
+
+    kwargs -- define addition keyworded arguemts for domain file. E.g. ln_sco=1
+    if using s-scoord in an old domain file that does not carry this flag.
+    """
+
+    def __init__(self, fn_data=None, fn_domain=None,  # TODO Super init not called + add a docstring
+                 multiple=False, config: str = " ",
+                 workers=2, threads=2, memory_limit_per_worker='2GB', **kwargs):
+        debug(f"Creating new {get_slug(self)}")
+        self.dataset = xr.Dataset()
+        self.grid_ref = None
+        self.domain_loaded = False
+        self.fn_data = fn_data
+        self.fn_domain = fn_domain
+        self.grid_vars = None
+
+        if path_lib.isfile(config):
+            self.config = ConfigParser(config).config
+        else:
+            error(f"Missing required json config :- {config} is not a file!")
+            raise FileNotFoundError
+
+        if self.config.chunks:
+            self._setup_grid_obj(self.config.chunks, multiple, **kwargs)
+        else:
+            self._setup_grid_obj(None, multiple, **kwargs)
+
+    def _setup_grid_obj(self, chunks, multiple, **kwargs):
+        """This is a helper method to reduce the size of def __init__
+
+        Args:
+            chunks: This is a setting for xarray as to whether dask (parrell processing) should be on and how it works
+            multiple: falg to tell if we are loading one or more files
+            **kwargs: pass direct to loaded xarray dataset
+        """
+        self.set_grid_vars()
+        self.set_dimension_mapping()
+        self.set_variable_mapping()
+
+        if self.fn_data is not None:
+            self.load(self.fn_data, chunks, multiple)
+
+        self.set_dimension_names(self.config.dataset.dimension_map)
+        self.set_variable_names(self.config.dataset.variable_map)
+
+        if self.fn_domain is None:
+            self.filename_domain = ""  # empty store for domain fileanme
+            warn("No NEMO domain specified, only limited functionality" +
+                 " will be available")
+        else:
+            self.filename_domain = self.fn_domain  # store domain fileanme
+            dataset_domain = self.load_domain(self.fn_domain, chunks)
+
+            # Define extra domain attributes using kwargs dictionary
+            ## This is a bit of a placeholder. Some domain/nemo files will have missing variables
+            for key, value in kwargs.items():
+                dataset_domain[key] = value
+
+            if self.fn_data is not None:
+                dataset_domain = self.trim_domain_size(dataset_domain)
+            self.set_timezero_depths(
+                dataset_domain)  # THIS ADDS TO dataset_domain. Should it be 'return'ed (as in trim_domain_size) or is implicit OK?
```

Comment:
I think we'd benefit from agreeing on a project-level line length convention and giving each module a Black pass before we commit.